src/pack.rs

```diff
@@ -258,44 +258,96 @@ async fn download_package(
     let file_name = package
         .file_name()
         .ok_or(anyhow!("could not get file name"))?;
-    let mut dest = File::create(output_dir.join(file_name)).await?;
-
-    tracing::debug!("Fetching package {}", package.url());
-    let mut response = client.get(package.url().to_string()).send().await?;
-    if response.status().is_client_error() {
-        return Err(anyhow!(
-            "failed to download {}: {}",
-            package.url().to_string(),
-            response.text().await?
-        ));
-    }
+    let package_url = package.url();
+
+    match package_url.scheme() {
+        "file" => {
+            // Handle file:// URLs
+            let local_path = package_url
+                .to_file_path()
+                .map_err(|_| anyhow!("Invalid file URL: {}", package_url))?;
+            tracing::debug!("Resolved local path: {}", local_path.display());
+
+            // Check file existence
+            if !local_path.exists() {
+                return Err(anyhow!(
+                    "File does not exist at path: {}",
+                    local_path.display()
+                ));
+            }
 
-    while let Some(chunk) = response.chunk().await? {
-        dest.write_all(&chunk).await?;
-    }
+            // Check file accessibility
+            if let Err(e) = tokio::fs::File::open(&local_path).await {
+                tracing::error!("Failed to open file {}: {:?}", local_path.display(), e);
+                return Err(anyhow!(
+                    "Could not access file at path {}: {:?}",
+                    local_path.display(),
+                    e
+                ));
+            }
+
+            // Ensure target directory exists
+            let target_path = output_dir.join(file_name);
+            create_dir_all(target_path.parent().unwrap_or_else(|| Path::new(".")))
```

Comment:
This seems fishy. Why do we need it and why using the current working directory as default? Shouldn't we rather fail in these cases?